app/src/main/java/me/hufman/androidautoidrive/phoneui/MusicQueuePageFragment.kt

```diff
@@ -0,0 +1,172 @@
+import android.content.Context
+import android.graphics.Bitmap
+import android.graphics.drawable.Animatable2
+import android.graphics.drawable.AnimatedVectorDrawable
+import android.graphics.drawable.Drawable
+import android.os.Bundle
+import android.os.Handler
+import android.support.v4.app.Fragment
+import android.support.v7.widget.LinearLayoutManager
+import android.support.v7.widget.RecyclerView
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.ImageView
+import android.widget.TextView
+import kotlinx.android.synthetic.main.music_queuepage.*
+import kotlinx.coroutines.*
+import me.hufman.androidautoidrive.R
+import me.hufman.androidautoidrive.music.MusicController
+import me.hufman.androidautoidrive.music.MusicMetadata
+import me.hufman.androidautoidrive.music.QueueMetadata
+import kotlin.collections.ArrayList
+import kotlin.coroutines.CoroutineContext
+
+class MusicQueuePageFragment: Fragment(), CoroutineScope {
+	override val coroutineContext: CoroutineContext
+		get() = Dispatchers.Main
+
+	companion object {
+		fun newInstance(): MusicQueuePageFragment {
+			val fragment = MusicQueuePageFragment()
+			fragment.arguments = Bundle().apply {
+
+			}
+			return fragment
+		}
+	}
+
+	lateinit var musicController: MusicController
+	lateinit var placeholderCoverArt: Bitmap
+
+	private var loaderJob: Job? = null
+	private val contents = ArrayList<MusicMetadata>()
+	private var currentQueueMetadata: QueueMetadata? = null
+
+	val handler = Handler()
+
+	fun onActive() {
+		musicController.listener = Runnable {
+			if(currentQueueMetadata?.title != musicController.getQueue()?.title || currentQueueMetadata?.songs?.size != musicController.getQueue()?.songs?.size) {
+				redrawQueueUI()
+			}
+
+			(this.context as Activity).listQueue.adapter?.notifyDataSetChanged()
+		}
+	}
+
+	override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+		return inflater.inflate(R.layout.music_queuepage, container, false)
+	}
+
+	override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+		val viewModel = ViewModelProviders.of(requireActivity()).get(MusicActivityModel::class.java)
+		val musicController = viewModel.musicController ?: return
+		this.musicController = musicController
+		placeholderCoverArt = viewModel.icons[MusicNowPlayingFragment.PLACEHOLDER_ID]!!
+
+		listQueue.setHasFixedSize(true)
+		listQueue.layoutManager = LinearLayoutManager(this.context)
+		listQueue.adapter = QueueAdapter(this.context!!, contents) { mediaEntry ->
+			if (mediaEntry != null) {
+				musicController.playQueue(mediaEntry)
+				(activity as MusicPlayerActivity).showNowPlaying()
+			}
+		}
+
+		listQueueRefresh.setOnRefreshListener {
+			redrawQueueUI()
+			Handler(this.context?.mainLooper).postDelayed({
+				listQueueRefresh.isRefreshing = false
+			}, 1000)
+		}
+
+		txtQueueEmpty.text = getString(R.string.MUSIC_BROWSE_LOADING)
```

Comment:
I believe the Android's MediaController getQueue api is synchronous, and so this should say that the queue is empty instead of loading. If a queue does take a bit to load, the MusicController callback should update this automatically.

Reply:
It is synchronous, fixed this to read that it is empty